### PR TITLE
[shopsys] added change and remove methods to js register class

### PR DESCRIPTION
**Pull request by pesektomas**
| Q             | A
| ------------- | ---
|Description, reason for the PR| If we want override  some js class (with constructor or init method) we can't use monkey patching, but we have to use standard classes overriding (like this class B extend A).  For this use case we have to change registered callback and this PR added method to change callback.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docs/frontend/npm-and-webpack.md [hunk 1]
```diff
@@ -143,3 +143,32 @@ CustomizeBundle.showFormErrorsWindow = myOverridedShowFormErrorsWindow;
 ```
 
 This principle is called [Monkey Patching](https://www.sitepoint.com/pragmatic-monkey-patching/).
+
+### I want to override class from @shopsys/framework common package
+
+You can use ES6 syntax for override class. You certainly know key word `extend`. You can use it in es world now.
```
Comment (grossmannmartin) [inline docs/frontend/npm-and-webpack.md]:
```suggestion
You can use ES6 syntax to override class.
You certainly know the key word `extend`.
You can use it in the ES world now.
```

File: packages/framework/assets/js/admin/validation/form/freeTransportAndPayment.js [hunk 2]
```diff
@@ -18,4 +18,4 @@ export default function validationFreTransportAndPayment () {
     });
 }
 
-(new Register()).registerCallback(validationFreTransportAndPayment);
+(new Register()).registerCallback(validationFreTransportAndPayment, 'validationFreTransportAndPayment');
```
Comment (grossmannmartin) [inline packages/framework/assets/js/admin/validation/form/freeTransportAndPayment.js]: here it looks like a typo (fre X free). I'm not sure if the class should be renamed in this PR, but the key can be correct even now, or?

File: packages/framework/assets/js/admin/validation/form/mailTemplate.js [hunk 3]
```diff
@@ -20,4 +20,4 @@ export default function validationMailTemplate () {
     });
 }
 
-(new Register()).registerCallback(validationMailTemplate);
+(new Register()).registerCallback(validationMailTemplate, 'mailTemplate.validationMailTemplate');
```
Comment (grossmannmartin) [inline packages/framework/assets/js/admin/validation/form/mailTemplate.js]: is `mailTemplate.validationMailTemplate` correct? For example `validationAdvert` uses just `validationAdvert` without any prefix

File: packages/framework/assets/js/admin/validation/form/freeTransportAndPayment.js (hunk 2 above)
Comment (pesektomas) [reply packages/framework/assets/js/admin/validation/form/freeTransportAndPayment.js]: 745c39a79812facb4e2922516390a1c9e137086a

File: packages/framework/assets/js/admin/validation/form/mailTemplate.js (hunk 3 above)
Comment (pesektomas) [reply packages/framework/assets/js/admin/validation/form/mailTemplate.js]: 1c9bd3f272d46c82caa4dc1e1722134ef74d13f8

File: docs/frontend/npm-and-webpack.md (hunk 1 above)
Comment (pesektomas) [reply docs/frontend/npm-and-webpack.md]: 7872665e959a169f7ab08abe7e64147a91d51fac

File: docs/frontend/npm-and-webpack.md [hunk 4]
```diff
@@ -143,3 +143,40 @@ CustomizeBundle.showFormErrorsWindow = myOverridedShowFormErrorsWindow;
...
+        console.log('patched constructor');
+    }
+
+    init () {
+        new MyCategoryTreeSorting();
+    }
+}
+```
+
+If you override framework's common javascript class, you will have to change registered callback with original class init.
```
Comment (grossmannmartin) [inline docs/frontend/npm-and-webpack.md]:
```suggestion
If you override framework's javascript class, you will have to change registered callback of the original class init to your implementation.
```

**Pull request merged**